packages/vaex-ml/vaex/ml/sklearn.py

```diff
@@ -95,3 +95,141 @@ def fit(self, df, target, **kwargs):
         X = df[self.features].values
         y = df.evaluate(target)
         self.model.fit(X=X, y=y, **kwargs)
+
+
+@vaex.serialize.register
+@generate.register
+class IncrementalPredictor(state.HasState):
+    '''This class wraps any scikit-learn estimator (a.k.a predictions) that has
+    a `.partial_fit` method, and makes it a vaex pipeline object.pipeline
+
+    By wrapping "on-line" scikit-learn estimators with this class, they becoes a vaex
+    pipeline object. Thus, they can take full advantage of the serialization and
+    pipeline system of vaex. While the underlying estimator need to call the
+    `.partial_fit` method, this class contains the standard `.fit` method, and
+    the rest happens behind the scenes. One can also iterate over the data
+    multiple times (epochs), and optionally shuffle each batch before it is sent
+    to the estimator. The `predict` method returns a numpy, while the `transform`
+    method adds the prediction as a virtual column to a vaex DataFrame.
```

Comment:
this is general docs, should not be put here right?

Reply:
Yeah, but I don't find it overly long, and it might be helpful to new users.

Reply:
But my point is that general info should not be put everywhere, but in one place to avoid maintenance burden, just an advice.